Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/Migration/Tools/MigrationTool.cs

```diff
@@ -0,0 +1,256 @@
+                    Debug.Log("d");
+                    return false;
+                }
+            }
+            var previousScenePath = EditorSceneManager.GetActiveScene().path;
+
+            foreach (var migrationObject in migrationObjects)
+            {
+                string assetPath = AssetDatabase.GetAssetPath(migrationObject);
+
+                // Object migrationObject is a scene Game Object
+                if (String.IsNullOrEmpty(assetPath))
+                {
+                    MigrateGameObjectHierarchy((GameObject)migrationObject);
+                }
+                else
+                {
+                    if (migrationObject is GameObject)
+                    {
+                        MigratePrefab(assetPath);
+                    }
+                    else if (migrationObject is SceneAsset)
+                    {
+                        MigrateScene(assetPath);
+                    }
+                }
+            }
+            migrationObjects.Clear();
+
+            if (!String.IsNullOrEmpty(previousScenePath) && previousScenePath != EditorSceneManager.GetActiveScene().path)
+            {
+                EditorSceneManager.OpenScene(Directory.GetCurrentDirectory() + "/" + previousScenePath);
+            }
+            return true;
+        }
+
+        private void MigrateScene(String path)
+        {
+            if (!AssetDatabase.LoadAssetAtPath(path, typeof(SceneAsset)))
+            {
+                return;
+            }
+            Scene scene = EditorSceneManager.OpenScene(path);
+
+            foreach (var parent in scene.GetRootGameObjects())
+            {
+                MigrateGameObjectHierarchy(parent);
+            }
+            EditorSceneManager.SaveScene(scene);
+        }
+
+        private void MigratePrefab(String path)
+        {
+            if (!AssetDatabase.LoadAssetAtPath(path, typeof(GameObject)))
+            {
+                return;
+            }
+            var parent = UnityEditor.PrefabUtility.LoadPrefabContents(path);
+
+            MigrateGameObjectHierarchy(parent);
+
+            UnityEditor.PrefabUtility.SaveAsPrefabAsset(parent, path);
+            PrefabUtility.UnloadPrefabContents(parent);
+        }
+
+        private void MigrateGameObjectHierarchy(GameObject parent)
+        {
+            if (!parent)
+            {
+                return;
+            }
+
+            foreach (var child in parent.GetComponentsInChildren<Transform>())
+            {
+                try
+                {
+                    if (migrationHandlerInstance.CanMigrate(child.gameObject))
+                    {
+                        migrationHandlerInstance.Migrate(child.gameObject);
+                        Debug.Log($"Successfully migrated {parent.name} object");
+                    }
+                }
+                catch (Exception e)
+                {
+                    Debug.LogError($"{e.Message}: GameObject {parent.name} could not be Migrated");
```

Comment:
```suggestion
                    Debug.LogError($"{e.Message}: GameObject {parent.name} could not be migrated");
```

Reply:
For Saving scene, the editor utility is poping window and asking if user wants to modify current scene, both options, Save scene or Do Not Save, return true. enough?